File: python/test_numeric.py
import pytest
import numeric as n

@pytest.fixture
def ary():
    return [1, 2, 3]

def test_itoa():
    g = n.itoa(1)
    assert 1 == g.next()
    assert 2 == g.next()

def test_accumulate(ary):
    assert n.accumulate(ary, 1, lambda x, y : x * y) == 6

def test_inner_product(ary):
    assert n.inner_production(ary, ary, 0) == 14

def test_adjacent_difference(ary):
    assert n.adjacent_difference(ary) == [1, 1, 1]

def test_partial_sum(ary):
    assert n.partial_sum(ary) == [1, 3, 6]

File: ruby/test_numeric.rb
require 'minitest/autorun'
require_relative 'numeric'

class TestNumeric < MiniTest::Test
  include NumericAlgorithm

  def setup
    @a = [1, 2, 3]
  end

  def test_itoa
    y = itoa(1)
    assert_equal 1, y.next
    assert_equal 2, y.next
  end

  def test_accumulate
    assert_equal 6, accumulate(@a, 1){|x, y| x * y}
  end

  def test_inner_product
    assert_equal 14, inner_product(@a, @a, 0)
  end

  def test_adjacent_difference
    assert_equal [1, 1, 1], adjacent_difference(@a)
  end

  def test_partial_sum
    assert_equal [1, 3, 6], partial_sum(@a)
  end
end
